### PR TITLE
Bump scalafmt from 3.6.1 to 3.7.1

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- Bump scalafmt from 3.6.1 to 3.7.1, release notes: https://github.com/scalameta/scalafmt/releases/tag/v3.7.1
- Changes to existing code seem coming from the dependency bump of scalameta 4.7.x in scalafmt 3.7.0+ (https://github.com/scalameta/scalafmt/releases/tag/v3.7.0) used for parsing scala code and looking up the nested constructor usages.
- Notice: `spotless-maven-plugin` is now pinned on 2.30.0 as from 2.31.0 requiring Java 11+, and 2.31.0 is using `scalafmt` 3.7.1 by default, see (https://github.com/diffplug/spotless/blob/main/plugin-maven/CHANGES.md#2310---2023-01-26)

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala [hunk 1]
```diff
@@ -45,8 +45,7 @@ private[v1] class AdminResource extends ApiRequestContext with Logging {
 
   @ApiResponse(
     responseCode = "200",
-    content = Array(new Content(
-      mediaType = MediaType.APPLICATION_JSON)),
+    content = Array(new Content(mediaType = MediaType.APPLICATION_JSON)),
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala:48]: Is it allowed to eliminate `Array()`?

Comment (bowenliang123) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala:48]: Seems not possible.

**Review by pan3793 — APPROVED** [review]

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala (hunk 1 above)
Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala:48]: Got it, it's fine then

**Comment by bowenliang123** [comment]
thanks, merged to master.

**Pull request closed without merging**